proposals/NNNN-Processing-of-unknown-enum-values-by-SDL-and-Mobile-libraries.md

```diff
@@ -0,0 +1,224 @@
+
+  ***Note** for SDL *PROTOCOL* version less than 4.3 in StartServiceACK response Proxy can get only *MAJOR* version*
+
+  ***Note** for SDL *PROTOCOL* version higher than or equal 4.4 in in StartServiceACK response Proxy can get *MAJOR*.*MINOR*.*PATCH* versions*
+
+  ***Note** since that moment (StartServiceACK response) and until RegisterAppInterface successful response Proxy should use PROTOCOL version to filter out all unknown enum values*
+
+  So the workflow should be implemented as follows:
+  1. The mobile application (Android/IOS) creates a proxy object (proxy) to communicate with SDL Core.
+  2. The mobile application passes to the Proxy entity the configurations.
+  3. Among other properties the configuration object has the following parameters:
+      * Minimal SDL version
+      * Application HMI type(s) (`AppHMIType` array)
+
+#### Changes related to sending request from mobile application to SDL
+  After the connection is established and mobile application is registered with SDL Core, mobile application works as usual but requires the request parameters validation. An issue will occur when the mobile application calls a method and passes an enum values array as a parameter that belongs to the higher SDL version. In this case, the Mobile Proxy should filter out unsupported params before sending the request and provide warning in logs as `unsupported_values`.
+
+  The validation sequence:
+
+  1. Mobile application sends request to SDL Core through the proxy.
+  2. The proxy checks the request parameters:
+     * If there are no enum values unknown to SDL, the proxy forwards the request to SDL Core without changes.
+
+     * If some of the enum values are unknown to SDL Core the proxy should:
+          * Filter out all RPC’s unsupported enum values including nested RPC’s if those are present, before making a request to make sure it is compliant with the current spec
+          * Log on warning level with «unsupported_values» (enum’s/struct’s names)
+
+  **Note**: For `RegisterAppInterface` to filter we should use version of PROTOCOL, but not the MOBILE_API according to system’s flow. **iOS** should be extended with similar mechanism in order to encapsulate filtering logic.
+
+  **For example** :
+  If the UI app requested unsupported `AppHMIType` and version which are not compatible, then the proxy must filter unsupported HMI types.
+
+   - Before filtering :
+       ```
+       "AppHMIType" : ["DEFAULT", "MEDIA", "UKNOWN"]
+       ```
+
+   - After filtering :
+       ```
+       AppHMIType" : ["DEFAULT", "MEDIA"]
+       ```
+
+  ***Note**: In case no HMI types are left after filtering, Mobile library should throw Exception, otherwise RegisterAppInterface RPC will be sent with remaining HMI types.*
```

Comment:
```suggestion
  ***Note**: In case no HMI types are left after filtering, Mobile Library should throw an exception, otherwise `RegisterAppInterface` RPC will be sent with remaining HMI types.*
```

Reply:
Backticks are still needed around "RegisterAppInterface" here.  Please update.

Reply:
@theresalech Done. Please take a look at the f24753617739884847fba10c9ff538da4c3e1b98